src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +155,85 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
                 IsBicep = isBicep,
-                SourceMap = sourceMap
+                SourceMap = sourceMap,
+                PathPrefix = pathPrefix
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
-                evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                if (this.powerShellRuleEngine != null)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
-                    {
-                        evalsToNotValidate.Add(eval);
-                    }
+                    this.logger?.LogDebug("Running PowerShell rule engine");
+                    evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
-                var uniqueResults = new Dictionary<(string, int), IEvaluation>();
-                foreach (var eval in evalsToValidate)
+               
+                // Recursively handle nested templates 
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(populatedTemplate);
+                dynamic processedTemplateResources = templatejObject["resources"];
+
+                for (int i = 0; i < processedTemplateResources.Count; i++)
                 {
-                    uniqueResults.TryAdd((eval.RuleId, eval.Result.LineNumber), eval);
-                }
-                evaluations = uniqueResults.Values.Concat(evalsToNotValidate);
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (currentProcessedResource.type == "Microsoft.Resources/deployments")
+                    {
+                        var nestedTemplate = currentProcessedResource.properties.template;
+                        if (nestedTemplate == null)
+                        {
+                            continue; // This is a Linked template 
+                        }
+                        dynamic populatedNestedTemplate = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(nestedTemplate));
 
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResource.properties.expressionEvaluationOptions?.scope;
+                        string nextPathPrefix = $"resources[{i}].properties.template.";
```

Comment:
After converting to use `JObject.Parse`, you can use the `JToken` object to access `.Path` instead of manually constructing the path to the nested template.

Reply:
I just realized that when I reconstruct it I do the mapping so copied resources map to their originals. If I use .Path it just gives me the unmapped resource path which still needs to be mapped.

Reply:
I ended up using this and moving all line numbering to ResolveLineNumbers as discussed